### DISCUSSION TITLE
Two external modules referencing different versions of the same library

### DISCUSSION THREAD

**Discussion by vzdesic** [oqtane/oqtane.framework#1375] [Q&A] [answered]
Is it possible to have two External Modules loaded into Oqtane, each of them using different version of the same external library?

For example, one using Radzen Blazor 3.3.0, another using Radzen Blazor 3.2.0?

Thanks, Vedran

**Comment by sbwalker** [comment, marked as answer]
I have my doubts on whether this would work well. It is not really an Oqtane issue - it is a .NET Core limitation. This scenario is precisely the reason why the high level Oqtane framework philosophy (https://www.oqtane.org/Resources/Blog/PostId/538/oqtane-philosophy) includes "Minimize Dependencies"

**Reply by vzdesic** [reply]
I think it is possible and it can work fine. There is an description how to load DLL dynamically: https://docs.microsoft.com/en-us/dotnet/api/system.appdomain.load?redirectedfrom=MSDN&view=net-5.0#overloads
But, it is not simple task, it would require additional organization of the DLLs inside the project, some predefined structure. Also, it can be probably developed independently, by leveraging ClientStartup mechanism.

**So, let's leave it for now!** 

Let me just explain why I did ask that. So, External Modules (many of them) developed in my system will communicate with many microservices below. With this feature I will be able to upgrade and deploy single module without necessary upgrade all External Modules at once. This is it.

Thanks a lot!
